src/Operation/DashPathSegmentNameGenerator.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Operation;
+
+use Doctrine\Common\Util\Inflector;
+
+/**
+ * Generate a path name with an underscore separator according to a string and whether it needs pluralization.
+ *
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class DashPathSegmentNameGenerator implements PathSegmentNameGeneratorInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getSegmentName(string $name, bool $pluralize = true): string
```

Comment:
I don't like the `$pluralize` argument. It should be up to the implementation to decide whether they want to pluralize or do other things. What we need to provide here is the context (e.g. whether this is for a collection or item).